src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This class holds ALL types of delegates that are coverted from ONE target method
+    /// Type parameters are generated for: <see cref="GetTypeArgumentsFromConversion(TypeSymbol, MethodSymbol)"/>
+    ///     1. All type parameters from the containing types of the target method
+    ///     2. All type parameters from the target method
+    ///     3. The delegate type
+    /// </summary>
+    /// <remarks>
+    /// This symbol is created in lowering phrase, and is collected and translated in emit phrase.
+    /// </remarks>
+    internal sealed class MethodGroupConversionCacheFrame : NamedTypeSymbol
```

Comment:
I think it would be much simpler if the cache is local -- that is it should be emitted into the class that contains the method body containing the conversion. 
For non-generic methods the field holding on the cached delegate can be emitted directly to the declaring type. No additional type is needed. Correct? For generic method let's emit the field into a nested generic class.

To name the field and the type use ```GeneratedNames.MakeMethodScopedSynthesizedName``` in the similar way it's used to generate name for ```MakeDynamicCallSiteContainerName``` -- that is pass in methodOrdinal and generation.